toolchain/strace.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM
+# Author: Sachin Sant <sachinp@linux.vnet.ibm.com>
+
+import os
+import re
+
+from avocado import Test
+from avocado.utils import build, process, git
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Strace(Test):
+
+    """
+    Strace is a diagnostic and debugging command line utility for Linux
+    :avocado: tags=os,testsuite
+    """
+
+    def setUp(self):
+        """
+        Build strace
+
+        Source:
+        http://github.com/strace/strace.git
+        """
+        smm = SoftwareManager()
+        packages = ['make', 'gcc', 'autoconf', 'automake']
+        for package in packages:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel(' %s is needed for the test to be run' % package)
+        git.get_repo('https://github.com/strace/strace.git',
```

Comment:
can we use zip file as in that case we can leverage  `fetch_asset`  API ?

Reply:
strace project hosted on github does not allow to download a zip/tarball, so I think this is not an option.

strace's program version is maintained using git-version-gen script, therefore, building strace depends on "git describe" (a full git clone is needed for this)or, alternatively, a .tarball-version file.  Neither github nor gitlab can
provide the latter, they also do not allow to inject a .tarball-version file during export archiving.